jekyll/_cci2/workflows.md

```diff
@@ -0,0 +1,205 @@
+
+![Fan-out and Fan-in Workflow]({{ site.baseurl }}/assets/img/docs/fan_in_out.png) 
+
+The following `config.yml` snippet is an example of a workflow configured for sequential job execution:
+
+``` 
+workflows:
+  version: 2
+  build_accept_deploy:
+    jobs:
+      - build:
+      - acceptance_test_1:
+          requires:
+            - build
+      - acceptance_test_2:
+          requires:
+            - build
+      - acceptance_test_3:
+          requires:
+            - build
+      - acceptance_test_4:
+          requires:
+            - build          
+      - deploy:
+          requires:
+            - acceptance_test_1
+            - acceptance_test_2
+            - acceptance_test_3
+            - acceptance_test_4
+```
+ 
+## Branch-Level Job Execution
+The following example shows a workflow configured with jobs on three branches: Dev, Stage, and Pre-Prod. 
+
+![Branch-Level Job Execution]({{ site.baseurl }}/assets/img/docs/branch_level.png) 
+
+The following `config.yml` snippet is an example of a workflow configured for branch-level job execution:
+
+```
+workflows:
+  version:2
+  dev_stage_pre-prod_master_deploy:
+    jobs:
+      - test_dev:
+          filters:
+            branches:
+              only: dev
+      - test_stage:
+          filters:
+            branches:
+              only: stage
+      - test_pre-prod:
+          filters:
+            branches:
+              only: pre-prod
+```
+
+## Workspaces
+
+Workflows that include jobs running on multiple branches may require artifacts to be shared using workspaces. To persist an artifact from a job and make it available to other jobs, configure the job to use the `persist_to_workspace` key where the value is a directory inside the project’s working directory. Artifacts of the job will be saved to this directory until the job is rerun and new artifacts are created.
+
+Configure a job to get a saved artifact by configuring the `attach_workspace` key where the value of the `at:` option is the directory defined in the `persist_to_workspace` key. The following `config.yml` file defines two jobs where the `downstream` job uses the artifact of the `flow` job. The workflow configuration is sequential, so that `downstream` requires `flow` to finish before it can start. 
+ 
+```
+defaults: &defaults
+  working_directory: /tmp
+  docker:
+    - image: buildpack-deps:jessie
+
+version: 2
+jobs:
+  flow:
+    <<: *defaults
+    steps:
+      - run: mkdir -p workspace
+      - run: echo "Hello, world!" > workspace/echo-output
+
+      - persist_to_workspace:
+          root: workspace
+          paths:
+            - echo-output
+
+  downstream:
+    <<: *defaults
+    steps:
+      - attach_workspace:
+          at: /tmp/workspace
+
+      - run: |
+          if [[ `cat workspace/echo-output` == "Hello, world!" ]]; then
```

Comment:
~ the command needs to be `cat /tmp/workspace/echo-output` because the workspace was attached to `/tmp/workspace`